File: Backend/Backend/GraphQlTypes/WordType.cs
using Backend.Repositories.Models;
using GraphQL.Types;

namespace Backend.GraphQlTypes
{
    public class WordType : ObjectGraphType<Word>
    {
        public WordType()
        {
            Field(x => x.Id).Description("Database id of this word.");
            Field(x => x.Name).Description("The name of this word.");
            Field(x => x.Translation).Description("The translation of this word.");
            Field(x => x.Grammar).Description("The type of grammar this word belongs to.");
            Field(x => x.Story).Description("The story to help remember this word.");
        }
    }
}

File: backend/src/graphql/types/wordType.js
import { GraphQLObjectType, GraphQLString, GraphQLNonNull } from 'graphql'

const wordType = new GraphQLObjectType({
    name: 'Word',
    description: 'A word',
    fields: {
        id: {
            type: new GraphQLNonNull(GraphQLString),
            description: 'The id of the word'
        },
        name: {
            type: new GraphQLNonNull(GraphQLString),
            description: 'The name of the word'
        },
        translation: {
            type: new GraphQLNonNull(GraphQLString),
            description: 'The translation of the word'
        },
        grammar: {
            type: GraphQLString,
            description: 'The grammar of the word'
        },
        story: {
            type: GraphQLString,
            description: 'The story of the word'
        }
    }
})

export default wordType
